specs/dynamicdependencies/DynamicDependencies.md

```diff
@@ -0,0 +1,1256 @@
+//
+// @param packageDependencyContext valid until passed to MddRemovePackageDependency()
+// @param packageFullName allocated via LocalAlloc; use LocalFree to deallocate
+STDAPI MddAddPackageDependency(
+    _In_ PCWSTR packageDependencyId,
+    INT32 rank,
+    MddMddAddPackageDependency flags,
+    _Out_ MDD_PACKAGEDEPENDENCY_CONTEXT* packageDependencyContext,
+    _Outptr_opt_result_maybenull_ PWSTR* packageFullName);
+
+// Remove a resolved PackageDependency from the current process' package graph
+// (i.e. undo MddAddPackageDependency). Used at runtime (i.e. the moral equivalent
+// of FreeLibrary).
+//
+// @note This does not unload loaded resources (DLLs etc). After removing
+//        a package dependency any files loaded from the package can continue
+//        to be used; future file resolution will fail to see the removed
+//        package dependency.
+STDAPI MddRemovePackageDependency(
+    _In_ MDD_PACKAGEDEPENDENCY_CONTEXT packageDependencyContext);
+
+// Return the package full name that would be used if the
+// PackageDependency were to be resolved. Does not add the
+// package to the process graph.
+//
+// @param packageFullName allocated via LocalAlloc; use LocalFree to deallocate
+STDAPI MddGetResolvedPackageFullNameForPackageDependency(
+    _In_ PCWSTR packageDependencyId,
+    _Outptr_result_maybenull_ PWSTR* packageFullName);
+
+// Return TRUE if PackageDependency pins would produce the same package
+// when resolved e.g. whether they share the same packageFamilyName,
+// minVersion, and packageDependencyProcessorArchitectures values.
+STDAPI_(BOOL) MddArePackageDependencyIdsEquivalent(
+    _In_ PCWSTR packageDependencyId1,
+    _In_ PCWSTR packageDependencyId2);
+
+// Return TRUE if packageDependencyIdContext1 and packageDependencyContextId2
+// are associated with the same resolved package.
+STDAPI_(BOOL) MddArePackageDependencyContextsEquivalent(
+    _In_ const MDD_PACKAGEDEPENDENCY_CONTEXT* packageDependencyContext1,
+    _In_ const MDD_PACKAGEDEPENDENCY_CONTEXT* packageDependencyContext2);
+```
+
+## 6.2. WinRT API
+
+```c# (but really MIDL3)
+namespace Microsoft.ApplicationModel
+{
+/// CPU architectures to optionally filter available packages against a package dependency.
+/// These generally correspond to processor architecture types supported by MSIX.
+/// @see Windows.System.ProcessorArchitecture
+enum PackageDependencyProcessorArchitectures
+{
+    None    = 0,
+    Neutral = 0x00000001,
+    X86     = 0x00000002,
+    X64     = 0x00000004,
+    Arm     = 0x00000008,
+    Arm64   = 0x00000010,
+    X86A64  = 0x00000020,
+};
+
+/// Options when 'pinning' a package dependency
+runtimeclass PinPackageDependencyOptions
+{
+    PinPackageDependencyOptions();
+
+    /// Optional filtering by cpu architecture(s)
+    PackageDependencyProcessorArchitectures Architectures;
+
+    /// Do not verify at least 1 matching package exists when pinning a package dependency
+    boolean DoNotVerifyDependencyResolution;
```

Comment:
Multiple-negatives are confusing.  Just say "ResolveDependencies" and it defaults to 'true' when constructed.